Sources/iOSClientExposureDownload/Download/SessionManager+Exposure.swift

```diff
@@ -440,26 +440,97 @@ extension iOSClientDownload.SessionManager where T == ExposureDownloadTask {
     }
     
     
-    /// Check if the license has expired or not
+    /// Check if the downloaded Asset has expired or not
     /// - Parameter assetId: asset id
     /// - Returns: true if the license has expired
-    public func isExpired(assetId: String)-> Bool {
+    public func isExpired(assetId: String, environment: Environment, sessionToken: SessionToken, completionHandler: @escaping (Bool) -> Void) {
         let downloadedAsset = getDownloadedAsset(assetId: assetId)
-        guard let playTokenExpiration = downloadedAsset?.entitlement?.playTokenExpiration else {
-            return true
+        
+        // Get download verified information
+        self.getDownloadVerified(assetId: assetId, environment: environment, sessionToken: sessionToken) { [weak self] verifiedInfo in
+            let publicationEndInMiliseconds = verifiedInfo?.publicationEnd.toDate()?.millisecondsSince1970
+            let playTokenExpirationInSeconds = downloadedAsset?.entitlement?.playTokenExpiration
+            
+            if let result = self?.calculateExpiry(publicationEndDateInMiliseconds: publicationEndInMiliseconds, playTokenExpirationInSeconds: playTokenExpirationInSeconds) {
+                completionHandler(result.1)
+            } else {
+                
+                // Calculation failed or did not return , assume downloaded asset is expired
```

Comment:
what if this method is called while network is unavailable? I.e. if the request to getDownloadVerified() fails because of disabled network, should we really be pessimistic?

Reply:
Very good question , It's up to us how to act on that scenario.  Should we let the `downloadedAsset` to be still valid or expired ?

Reply:
@udayasri I found similar issues with Ramanjot's implementation. Let's take this discussion on slack.
But I'm leaning on minimal changes w.r.t. previous solution (without publicationEnd).

LGTM otherwise.